packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -197,7 +197,9 @@ async function handleRequest(
 	const url = new URL(origin + req.url?.replace(/(index)?\.html$/, ''));
 	const pathname = decodeURI(url.pathname);
 	const rootRelativeUrl = pathname.substring(devRoot.length - 1);
-	if (!buildingToSSR) {
+
+	// HACK! @astrojs/image uses query params for the injected route in `dev`
+	if (!buildingToSSR && rootRelativeUrl !== '/_image') {
```

Comment:
Not super happy with this one... Long term it might be nice to have a `devOnly` flag for `injectRoute`

Other short term solutions considered:

- allow all endpoints to use query params in SSG
- allow routes beginning with `/_` to use query params, assuming they'd all be from an `injectRoute` call

Reply:
This is fine for now. I have some ideas for a follow-up but want you to be able to get this out there!